stable_baselines/common/env_checker.py

```diff
@@ -0,0 +1,218 @@
+    when the observation is apparently an image.
+    """
+    if observation_space.dtype != np.uint8:
+        warnings.warn("It seems that your observation is an image but the `dtype` "
+                      "of your observation_space is not `np.uint8`. "
+                      "If your observation is not an image, we recommend you to flatten the observation "
+                      "to have only a 1D vector")
+
+    if np.any(observation_space.low != 0) or np.any(observation_space.high != 255):
+        warnings.warn("It seems that your observation space is an image but the "
+                      "high and lower bounds are not in [0, 255]. "
+                      "Because the CNN policy normalize automatically the observation "
+                      "you may encounter issue if the values are not in that range."
+                    )
+
+    if observation_space.shape[0] < 36 or observation_space.shape[1] < 36:
+        warnings.warn("The minimal resolution for an image is 36x36 for the default CnnPolicy. "
+                      "You might need to use a custom `cnn_extractor` "
+                      "cf https://stable-baselines.readthedocs.io/en/master/guide/custom_policy.html")
+
+
+def _check_unsupported_obs_spaces(env: gym.Env, observation_space: spaces.Space):
+    """Emit warnings when the observation space used is not supported by Stable-Baselines."""
+
+    if isinstance(observation_space, spaces.Dict) and not isinstance(env, gym.GoalEnv):
+        warnings.warn("The observation space is a Dict but the environment is not a gym.GoalEnv "
+                      "(cf https://github.com/openai/gym/blob/master/gym/core.py), "
+                      "this is currently not supported by Stable Baselines "
+                      "(cf https://github.com/hill-a/stable-baselines/issues/133), you will need to use a custom policy. "
+                    )
+
+    if isinstance(observation_space, spaces.Tuple):
+        warnings.warn("The observation space is a Tuple,"
+                      "this is currently not supported by Stable Baselines "
+                      "(cf https://github.com/hill-a/stable-baselines/issues/133), "
+                      "you will need to flatten the observation and maybe use a custom policy. "
+                    )
+
+
+def _check_nan(env: gym.Env):
+    """Check for Inf and NaN using the VecWrapper."""
+    vec_env = VecCheckNan(DummyVecEnv([lambda: env]))
+    for _ in range(10):
+        action = [env.action_space.sample()]
+        _, _, _, _ = vec_env.step(action)
+
+
+def _check_obs(obs: Union[tuple, dict, np.ndarray, int],
+               observation_space: spaces.Space,
+               method_name: str):
+    """
+    Check that the observation returned by the environment
+    correspond to the declared one.
+    """
+    # Check for the reset method only: the step method return a tuple of 4 elements
+    if not isinstance(observation_space, spaces.Tuple) and method_name == 'reset':
```

Comment:
I don't understand this check. Isn't `_check_obs()` only called with `obs` not the whole tuple from `step`?

Reply:
The check is still valid for `step()` but the error message would be misleading in that case (because `step()` returns a tuple)

Reply:
I'd favour changing the error message then, e.g. `"The observation returned by the `{}()` method should be a single value, not a tuple"`. But I do not feel that strongly about this.